src/processing.rs

```diff
@@ -80,9 +91,9 @@ pub fn stack_frames(
     );
 
     // Initial necessary values
-    let length_signal = sig.shape[0];
-    let frame_sample_length = (sampling_frequency * frame_length).round(); // Defined by the number of samples
-    let frame_stride = (sampling_frequency * frame_stride).round();
+    let length_signal = sig.len();
```

Comment:
Danger! len() doesn't return the shape but rather the total number of elements! If we have a 3x2 matrix, we will get 6! For one dimensional arrays should be fine, just let's keep this in mind.